### ISSUE TITLE
IOS background location tracking stops after 40 minutes.

### ISSUE THREAD

**Issue by Akhidha00** [michalchudziak/react-native-geolocation#314]
I want to track the device for more than 10 hours in IOS background(app is in minimised state).But the tracking ends in 40 minutes.
Is there any solution for this?
The code is:
await Geolocation.watchPosition(
      (position) => {
        var userLat = position.coords.latitude;
        var userLon = position.coords.longitude;
        if(latitude != userLat && longitude != userLon){
           getLocationAddress(userLat, userLon);
        }
      },
      (error) => {
        checkPermission()
      },
      { enableHighAccuracy: true, distanceFilter:500, fastestInterval: 300000, showsBackgroundLocationIndicator: true} // Optional configuration
    );

I am stuck with this.

**Comment by megabeehothanhminh** [comment]
the same issue

**Comment by Akhidha00** [comment]
Ios prevents the background task after some time.Thats why this happens.

**Comment by PhamQuyet1606** [comment]
the same issue, do you have any suggest for other libraries? @Akhidha00

**Comment by Akhidha00** [comment]
The `react-native-background-geolocation` library is good for tracking. It is free for iOS live builds, but requires payment for Android live builds. You can use the library in Android debug mode, but not in live mode without a paid license.

**Comment by petertoth-dev** [comment]
`react-native-background-geolocation` is a terrible package with a really rude maintainer with no real support. If you ask a question you get bullied instead of an answer. 
The whole package is a huge misunderstanding and misusing fitness permission for location tracking purposes. It slowed down our app, made a huge negative attempt on it. Avoid it at all costs if you listen to me...

**Comment by wmonecke** [comment]
@petertoth-dev I have interacted with the dev and he is not rude, he just gets to the point and probably has to deal with a lot of unexperienced devs just trying out the lib without reading the docs.

**Comment by petertoth-dev** [comment]
@wmonecke yes, so he is rude... if he has no time to communicate properly, hire someone who does... he is breaking the law with his "non refund" policy as well, he sells a half-ready, non-performant product for a price of a Jetbrains software license :)))  It's a scam.
